rewrite-java/src/main/java/org/openrewrite/java/UsePrecompiledRegExpForReplaceAll.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java;
+
+import org.openrewrite.Cursor;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.Space;
+import org.openrewrite.marker.Markers;
+
+import java.util.regex.Pattern;
+
+import static org.openrewrite.Tree.randomId;
+
+public class UsePrecompiledRegExpForReplaceAll extends Recipe {
+
+    public static final String REG_EXP_KEY = "regExp";
```

Comment:
How can VariableNameUtils help me with the key for the message I use to transport the actual regular expression up to the class level?

Reply:
I see you are already using `VariableNameUtils` on line 99, which is good. It would be nice if we at some point can figure out a strategy to get better variable names, as for example IDEA does it. But that is out of scope for this PR.